glean-core/ffi/src/timing_distribution.rs

```diff
@@ -0,0 +1,126 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::convert::TryFrom;
+use std::os::raw::c_char;
+
+use ffi_support::{define_handle_map_deleter, ConcurrentHandleMap, FfiStr};
+use lazy_static::lazy_static;
+
+use glean_core::metrics::{MetricType, TimeUnit, TimerId, TimingDistributionMetric};
+use glean_core::{CommonMetricData, Lifetime};
+
+use crate::handlemap_ext::HandleMapExtension;
+use crate::{from_raw_string_array, RawStringArray, GLEAN};
+
+lazy_static! {
+    pub static ref TIMING_DISTRIBUTION_METRICS: ConcurrentHandleMap<TimingDistributionMetric> =
+        ConcurrentHandleMap::new();
+}
+define_handle_map_deleter!(
+    TIMING_DISTRIBUTION_METRICS,
+    glean_destroy_timing_distribution_metric
+);
+
+#[no_mangle]
+pub extern "C" fn glean_new_timing_distribution_metric(
```

Comment:
nit: you should consider merging the macro and then rebasing to get rid of this :)